tools/common-code/cmonitor_watcher.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/python3
+
+#
+# cmonitor_watcher.py
+#
+# Author: Satyabrata Bharati
+# Created: April 2022
+#
+import inotify.adapters
+import queue
+import os
+import time
+import logging
+from datetime import datetime
+
+exit_flag = False
+# =======================================================================================================
+# CgroupWatcher : Basic inotify class
```

Comment:
this class looks good except for one thing: the timeout / sleep logic: it should not be included in this class.. it doesn't really belong here. The caller may want to sleep for whatever reason by watching cgroup should be fast and imply no sleeps. So please remove the "timeout" field.